etsi_its_conversion/etsi_its_cam_conversion/include/etsi_its_cam_conversion/convertCamParameters.h

```diff
@@ -43,39 +22,31 @@ namespace cam_msgs = etsi_its_cam_msgs::msg;
 namespace etsi_its_cam_conversion {
 
 void toRos_CamParameters(const CamParameters_t& in, cam_msgs::CamParameters& out) {
-
   toRos_BasicContainer(in.basicContainer, out.basic_container);
   toRos_HighFrequencyContainer(in.highFrequencyContainer, out.high_frequency_container);
   if (in.lowFrequencyContainer) {
     toRos_LowFrequencyContainer(*in.lowFrequencyContainer, out.low_frequency_container);
     out.low_frequency_container_is_present = true;
   }
-
   if (in.specialVehicleContainer) {
     toRos_SpecialVehicleContainer(*in.specialVehicleContainer, out.special_vehicle_container);
     out.special_vehicle_container_is_present = true;
   }
-
 }
 
 void toStruct_CamParameters(const cam_msgs::CamParameters& in, CamParameters_t& out) {
-
   memset(&out, 0, sizeof(CamParameters_t));
 
   toStruct_BasicContainer(in.basic_container, out.basicContainer);
   toStruct_HighFrequencyContainer(in.high_frequency_container, out.highFrequencyContainer);
   if (in.low_frequency_container_is_present) {
-    LowFrequencyContainer_t low_frequency_container;
-    toStruct_LowFrequencyContainer(in.low_frequency_container, low_frequency_container);
-    out.lowFrequencyContainer = new LowFrequencyContainer_t(low_frequency_container);
+    out.lowFrequencyContainer = (LowFrequencyContainer_t*) calloc(1, sizeof(LowFrequencyContainer_t));
+    toStruct_LowFrequencyContainer(in.low_frequency_container, *out.lowFrequencyContainer);
```

Comment:
I'm guessing there are memory leaks for messages requiring dynamic allocation (messages with optional fields, sequences ofs) in the current conversion node code, as there are no matching `delete`s for the `new`s. 
`asn1c` provides methods to [deep free structs](https://github.com/mouse07410/asn1c/blob/5fa129cf3161da9ff2d67341e6277e818c0f5d6a/skeletons/constr_TYPE.h#L85), however these methods use [by default](https://github.com/mouse07410/asn1c/blob/5fa129cf3161da9ff2d67341e6277e818c0f5d6a/skeletons/asn_internal.h#L40) `free()`.
The new `calloc` here should allow us to the cleanup methods like `ASN_STRUCT_FREE` or `ASN_STRUCT_FREE_CONTENTS_ONLY`.

Reply:
Good catch, unfortunately, we are already aware of that memory leak. We were considering wrapping the structs in yet another type providing a destructor for cleanup.

However, we didn't know about the existing methods for deep freeing structs yet, thanks! Does that mean that we can simply call those with a `CAM_t`? If yes, we should also integrate that into the conversion node to avoid the memory from leaking there.

Reply:
Yeah, it is easy to integrate. I can do it either on this PR or create another whenever this one is accepted.